src/tests/blueprint/t_blueprint_mesh_verify.cpp

```diff
@@ -52,20 +52,292 @@
 #include "blueprint.hpp"
 #include "relay.hpp"
 
-#include <iostream>
+#include <vector>
+#include <string>
 #include "gtest/gtest.h"
 
 using namespace conduit;
 
+/// Helper Functions ///
+
+std::vector<std::string> create_basis(const std::string& d1,
+                                      const std::string& d2,
+                                      const std::string& d3="")
+{
+    std::vector<std::string> dim_vector;
+
+    dim_vector.push_back(d1);
+    dim_vector.push_back(d2);
+
+    if(d3 != "")
+    {
+        dim_vector.push_back(d3);
+    }
+
+    return dim_vector;
+}
+
+
+bool is_valid_basis(bool (*basis_valid_fun)(const Node&, Node&),
+                    const std::vector<std::string>& basis)
+{
+    Node n, info;
+
+    bool is_valid = true;
+    for(index_t bi = 0; bi < basis.size(); bi++)
+    {
+        const std::string& basis_dim = basis[bi];
+
+        n[basis_dim].set("test");
+        is_valid &= !basis_valid_fun(n, info);
+
+        n[basis_dim].set(10);
+        is_valid &= basis_valid_fun(n, info);
+
+        // TODO(JRC): Determine whether or not the basis (i, k) should be
+        // valid for logical coordinates.
+        /*
+        if( bi > 0 )
+        {
+            const std::string& prev_dim = basis[bi-1];
+            n.remove(prev_dim);
+            is_valid &= !basis_valid_fun(n, info);
+            n[basis_dim].set(10);
+        }
+        */
+    }
+
+    return is_valid;
+}
+
+/// Testing Constants ///
+
+const std::vector<std::string> LOGICAL_BASIS = create_basis("i","j","k");
+const std::vector<std::string> CARTESIAN_BASIS = create_basis("x","y","z");
+const std::vector<std::string> SPHERICAL_BASIS = create_basis("r","theta","phi");
+const std::vector<std::string> CYLINDRICAL_BASIS = create_basis("r","z");
+
```

Comment:
I think coordinate system is good (the tuples in a coord sets are anchored to some coord system)